org.eclipse.lsp4e/src/org/eclipse/lsp4e/operations/symbols/LSPSymbolInFileDialog.java

```diff
@@ -113,12 +173,23 @@ protected Control createDialogArea(Composite parent) {
 				} catch (BadLocationException e) {
 					LanguageServerPlugin.logError(e);
 				}
-				close();
 			}
-		});
 
-		viewer.setInput(outlineViewerInput);
-		return filteredTree;
+			close();
+		}
+	}
+
+	protected Object getSelectedElement() {
+		if (treeViewer == null) {
+			return null;
+		}
+
+		IStructuredSelection selection = treeViewer.getStructuredSelection();
```

Comment:
I am not sure if treeViewer could be made asynchronously null (since it is now a class field), to be sure that we do not throw an NPE, I would assign if first to a local variable, then check the local variable, and if not null, continue using that local variable